libs/causality/src/lib/CausalAnalysisDashboard/Controls/CausalAnalysisView/TreatmentView/TreatmentView.tsx

```diff
@@ -71,9 +73,28 @@ export class TreatmentView extends React.Component<
     );
   }
 
+  public componentDidUpdate(prevProps: ITreatmentViewProps): void {
+    if (
+      prevProps.data &&
+      this.props.data &&
+      prevProps.data[0].local_policies &&
+      this.props.data[0].local_policies
+    ) {
+      if (
+        prevProps.data[0].local_policies.length !==
+        this.props.data[0].local_policies.length
```

Comment:
Could there be a case where length is same but the policies have been updated?
can we use _.isequal(prevprops, currentprops) in that case?
Why is it always comparing with 0th index in prevProps.data[0]?